src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -267,6 +271,18 @@ std::string GeneratedEnumFileName(const EnumDescriptor* en,
   return result + ".php";
 }
 
+std::string GeneratedServiceFileName(const ServiceDescriptor* service,
```

Comment:
Please reference GeneratedMessageFileName to implement this function.
Recently, we add php_namespace to specify custom namespace instead of using package.
We also have some logic to escape reserved name in php.